### PR TITLE
Ip adress sanitization for MP Menu screen

### PR DESCRIPTION
**Pull request by pGlase**
After being annoyed by Copy&Paste errors when copying from Discord i figured I could add this as an entry exercise.
Maybe this is a good idea to prevent A) Crashes on invalid IPs and B) Give at least a little feedback about IPv4 only.

Please let me know if this is useable for the project, I am usually a C++ dev.

### CODE REVIEW

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs [hunk 1]
```diff
@@ -171,6 +171,18 @@ private static void OnJoinGameButtonClick()
             string ip = parts[0];
             int port;
 
+            //remove copy and paste mistakes and update the textbox to prevent user confusion in case of invalid ip address
+            ip = SanitizeIpAddressString(ip);
```
Comment (Mushroom) [inline NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:175]: I think using `String.Trim()` is probably better than a new function here

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs [hunk 2]
```diff
@@ -171,6 +171,18 @@ private static void OnJoinGameButtonClick()
...
+            hostIPAdressInput.text = parts.Length == 1 ? ip : ip + ":" + parts[1];
+
+            if (!System.Net.IPAddress.TryParse(ip, out _))
+            {
+                //log, notify user and early return to prevent crash on invalid strings
+                InGamePopup.ShowWarning("Invalid connection string", "Note: This mod currently only accepts IPv4 Addresses.", "OK");
```
Comment (Mushroom) [inline NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:181]: Make sure the text in this message box is not clipped on low resolutions / high scaling, it's quite long

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs [hunk 3]
```diff
@@ -181,6 +193,7 @@ private static void OnJoinGameButtonClick()
                 return;
             }
 
+
```
Comment (Mushroom) [inline NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:196]: Random newline?

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs [hunk 4]
```diff
@@ -195,5 +208,18 @@ private static void OnJoinGameBackButtonClick()
             multiplayerMenu.gameObject.SetActive(false);
             UIRoot.instance.OpenMainMenuUI();
         }
+
+        private static string SanitizeIpAddressString(string ipAddr)
```
Comment (Mushroom) [inline NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:212]: See comment above about using `String.Trim()` instead

**Review by Mushroom — CHANGES_REQUESTED** [review]
Ahh, as an embedded dev I understand your approach completely, but this is not the C# way :)

I have suggested a better alternative

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs (hunk 2 above)
Comment (pGlase) [reply NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:181]: I just tried this on 640x480 on my 1440p display, i don't recommend it, but at least it isn't clipping ;)
grafik

Comment (Mushroom) [reply NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:181]: Alright then, that's one concern alleviated!

**Review by Mushroom — APPROVED** [review]

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs (hunk 1 above)
Comment (Therzok) [reply NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs]: Trim is identical to Trim(whitespace)

File: NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs [hunk 5]
```diff
@@ -171,6 +171,18 @@ private static void OnJoinGameButtonClick()
             string ip = parts[0];
             int port;
 
+            //remove copy and paste mistakes and update the textbox to prevent user confusion in case of invalid ip address
+            ip = ip.Trim('\r', '\n', '\t', ' ');
```
Comment (Therzok) [inline NebulaPatcher/Patches/Dynamic/UIMainMenu_Patch.cs:175]: These are all already considered whitespace by c#

**Pull request merged**